### PR TITLE
Improve errors and experience when referencing WinMDs

### LINKED ISSUE #14942: Reconsider implementation of _BlockWinMDsOnUnsupportedTFMs target
With the removal of WinRT support from .NET 5, the .NET SDK added the target _BlockWinMDsOnUnsupportedTFMs to ensure that winmd references were no longer passed to the C# compiler when targeting .NET 5+.  At the time this target was implemented, it was too late in the product cycle to address this in the C# compiler itself.  Unfortunately, there are many scenarios where a winmd reference can sneak into a project, causing a build break and presenting the sometimes misleading message:
NETSDK1130: Referencing a Windows Metadata component directly when targeting {0} is not supported. Use the C#/WinRT projection tool (https://aka.ms/cswinrt) or a provided projection for this target.

I recommend modifying the target to instead:

- Ignore and remove winmd references during compilation, to prevent use of WinRT types that will fail at runtime.  
- Issue a warning (not an error) that winmd references are no longer supported, possibly mentioning cswinrt if build errors do occur.

```
    <Target Name="_BlockWinMDsOnUnsupportedTFMs"
            AfterTargets="PreBuildEvent"
            DependsOnTargets="ResolveReferences"
            Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '5.0'))">
        <!--Winmdobj generation is never supported-->
        <NETSdkError Condition="'$(OutputType)' == 'winmdobj'"
          ResourceName="WinMDObjNotSupportedOnTargetFramework"
          FormatArguments="$(TargetFrameworkMoniker)" />
        <!--Winmd references should be removed, rather than failing the build-->
        <Warning Condition="'%(ReferencePath.Extension)' == '.winmd'"
            Text="Windows Metadata component references are ignored when targeting $(TargetFrameworkMoniker)."/>
        <ItemGroup>
            <WinMDReferencePath Include="@(ReferencePath)" Condition="'%(ReferencePath.Extension)' == '.winmd'" />
            <ReferencePath Remove="@(WinMDReferencePath)"/>
        </ItemGroup>
    </Target>

```

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #14942

- Updates the error generated when a WinMD file is referenced.  Checks if there are managed references with `windowsruntime` references, and if so errors about those references rather than the WinMDs.  Also updates the error message to list the filenames of the WinMDs that are referenced
- Removes WinMD references from referenced projects if they have Implementation metadata with a value of `WinRT.Host.dll`.  This fixes references to managed WinRT components, which include both the managed DLL and the WinMD in the target paths that the project returns.
- Adds test cases to cover various WinMD scenarios

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs [hunk 1]
```diff
@@ -32,41 +33,252 @@ public void It_fails_to_produce_winmds_for_net5_0_or_newer()
...
+    static void Main(string[] args)
+    {
+        Console.WriteLine(new ManagedWinRTComponent.Coord().ToString());
+    }
+}";
+
+            var testAsset = _testAssetsManager.CreateTestProject(consoleApp);
+
+            //  Disable workaround for NETSDK1130 which is in Microsoft.Windows.CsWinRT
+            File.WriteAllText(Path.Combine(testAsset.TestRoot, "Directory.Build.targets"), @"<Project>
```
Comment (j0shuams) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs:179]: I can turn it off once this PR merges and you won't have to do this I think...

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs [hunk 2]
```diff
@@ -32,41 +33,252 @@ public void It_fails_to_produce_winmds_for_net5_0_or_newer()
...
+            {
+                Name = "ManagedWinRTComponent",
+                TargetFrameworks = "net5.0-windows10.0.19041.0",
+            };
+
+            managedWinRTComponent.AdditionalProperties.Add("CsWinRTWindowsMetadata", "10.0.19041.0");
+            managedWinRTComponent.AdditionalProperties.Add("CsWinRTComponent", "true");
+            managedWinRTComponent.AdditionalProperties.Add("PlatformTarget", "x64");
+
+            //  TODO: Update to latest (currently 1.2.5) once it shows up on dotnet-public feed
```
Comment (sfoslund) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs:115]: Is this being tracked in an issue?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs [hunk 3]
```diff
@@ -32,41 +33,252 @@ public void It_fails_to_produce_winmds_for_net5_0_or_newer()
...
+            var projectionProject = new TestProject("SimpleMathProjection")
+            {
+                TargetFrameworks = "net5.0-windows10.0.19041.0"
+            };
+
+            projectionProject.AdditionalProperties["CsWinRTIncludes"] = "SimpleMathComponent";
+            projectionProject.AdditionalProperties["CsWinRTGeneratedFilesDir"] = "$(OutDir)";
+            projectionProject.AdditionalProperties["CsWinRTWindowsMetadata"] = "10.0.19041.0";
+
+            //  TODO: Update to latest version
```
Comment (sfoslund) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsRuntimeComponent.cs:220]: This TODO as well

**Review by sfoslund — APPROVED** [review]
LGTM

**Comment by sfoslund** [comment]
@dsplaisted I'm merging so we can get a build to insert for prev 3 today.

**Pull request merged**